File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let table = {};
    for (let i = 0; i < nums.length; i++) {
        if (table[target - nums[i]] !== undefined) {
            return [table[target - nums[i]], i];
        } else {
            table[nums[i]] = i;
    }
  }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        set = {}
        for i, num in enumerate(nums):
            if target - num in set:
                return [set.get(target - num), i] 
            else:
                set.setdefault(num, i)
        return []